### PR TITLE
Add Korean translation

### PR DESCRIPTION
**Pull request by bagjunggyu**
**Short description of changes**

This is creating translation_ko_KR.ts file for Korean.

CHANGELOG: Translation: Added Korean translation to Jamulus

**Context: Fixes an issue?**

https://github.com/jamulussoftware/jamulus/discussions/2653

**Does this change need documentation? What needs to be documented and how?**

No

**Status of this Pull Request**

Ready for testing

**What is missing until this pull request can be merged?**
Testing

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [x] I tested my code and it does what I want
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#source-code-consistency) 
-  [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [x] I've filled all the content above

### CODE REVIEW

**Comment by ann0see** [comment]
Hi! Great and big thanks for the translation. Around Line 108 there is a syntax error. You need to close the message tag there: `</message>`. 
I'm not sure if we need to add anything to Jamulus.pro, but probably yes. 

Did you already read https://github.com/jamulussoftware/jamulus/blob/master/docs/TRANSLATING.md?

**Comment by bagjunggyu** [comment]
OK, Thanks.

File: src/translation/translation_ko_KR.ts [hunk 1]
```diff
@@ -0,0 +1,4164 @@
...
+    </message>
+    <message>
+        <location filename="../clientsettingsdlgbase.ui" line="227"/>
+        <source>Meter Style</source>
+        <translation>미터 스타일</translation>
+    </message>
+    <message>
+        <location filename="../clientsettingsdlgbase.ui" line="241"/>
+        <source>Mixer Rows</source>
+        <translation믹서 줄</translation>
```
Comment (ignotus666) [inline src/translation/translation_ko_KR.ts:2297]: There's a ">" character missing after "translation".

Are you directly editing the .ts file? These errors are easily avoided if you use QtLinguist (or any other editor that can handle .ts files) instead. The problem is, if there are more of these syntax errors, this file might not load properly in QtLinguist, so no point switching now...

Comment (bagjunggyu) [reply src/translation/translation_ko_KR.ts:2297]: yes, I do.
I understand it, Thank you.

File: src/translation/translation_ko_KR.ts [hunk 2]
```diff
@@ -0,0 +1,4164 @@
...
+        <source>Cello</source>
+        <translation type="vanished">첼로</translation>
+    </message>
+    <message>
+        <source>Double Bass</source>
+        <translation type="vanished">콘트라베이스</translation>
+    </message>
+    <message>
+        <source>Recorder</source>
+        <translation type="vanished"리코더</translation>
```
Comment (ignotus666) [inline src/translation/translation_ko_KR.ts:2918]: ">" missing after "vanished".

Comment (bagjunggyu) [reply src/translation/translation_ko_KR.ts:2918]: I found some of them after that I would update at once.

Comment (ignotus666) [reply src/translation/translation_ko_KR.ts:2918]: If after summitting a correction you see checks at the bottom failing, click on 'Details' and you will see a log . Look for the line starting with 'lrelease error' near the bottom and it'll tell you what line the error is on.

Comment (bagjunggyu) [reply src/translation/translation_ko_KR.ts:2918]: Thank you :)

**Comment by bagjunggyu** [comment]
Actually it is derived from 'ending character error, or bug' in Input-Method for Korean.
I should have checked it deeply.

File: src/translation/translation_ko_KR.ts (hunk 2 above)
Comment (bagjunggyu) [reply src/translation/translation_ko_KR.ts]: It seems ok now.

**Comment by ignotus666** [comment]
Ok, no more errors ;) I can't comment on the Korean though...

**Comment by ann0see** [comment]
Yes. That's probably up to the other Korean translator(s).
Next step here:
* Add Korean translation to Jamulus.pro if needed
* Get the compiled version from t checks tab and test if the Korean translation shows up correctly
* Add involved translators to the translators tab in the app
* Add translators to GitHub organisation
* Add invite to DC/Matrix

**Comment by ignotus666** [comment]
@bagjunggyu I just opened your .ts file in QtLinguist and it shows a lot of untranslated strings, when they are actually translated. This is because the `type="unfinished"` tag is still there (it should be removed). Again, this is due to editing the file directly

**Comment by bagjunggyu** [comment]
@ignotus666  Yeah, That was my question https://github.com/jamulussoftware/jamulus/discussions/2683
OK, I shall do it right now. and Commit again soon.

**Comment by ann0see** [comment]
Ok. Compiled and included all the necessary files in Jamulus.pro and the qrc file in src. Now there's an error (to me unreadable):

QString::arg: 1 argument(s) missing in 왼쪽 및 오른쪽 로컬 오디오 채널 사이의 레벨을 적절하게 제어합니다. 모노 신호의 경우 두 채널 사이의 팬 역할을 합니다. 예를 들어 마이크가 오른쪽 입력 채널에 연결되어 있고 악기가 마이크보다 훨씬 큰 왼쪽 입력 채널에 연결된 경우 페이더 위의 레이블이 %1을 표시하는 방향으로 오디오 페이더를 이동합니다. 여기서 % 2는 현재 감쇠 표시기입니다.

**Comment by bagjunggyu** [comment]
@ann0see That line is
<message>
        <location filename="../clientsettingsdlg.cpp" line="59"/>
        <source>Controls the relative levels of the left and right local audio channels. For a mono signal it acts as a pan between the two channels. For example, if a microphone is connected to the right input channel and an instrument is connected to the left input channel which is much louder than the microphone, move the audio fader in a direction where the label above the fader shows %1, where %2 is the current attenuation indicator.</source>
        <translation>왼쪽 및 오른쪽 로컬 오디오 채널 사이의 레벨을 적절하게 제어합니다. 모노 신호의 경우 두 채널 사이의 팬 역할을 합니다. 예를 들어 마이크가 오른쪽 입력 채널에 연결되어 있고 악기가 마이크보다 훨씬 큰 왼쪽 입력 채널에 연결된 경우 페이더 위의 레이블이 %1을 표시하는 방향으로 오디오 페이더를 이동합니다. 여기서 % 2는 현재 감쇠 표시기입니다.</translation>
    </message>

**Comment by ann0see** [comment]
Ok. Wait a moment, I've just added my additions to your branch

**Comment by bagjunggyu** [comment]
OK, I understand it.

**Comment by bagjunggyu** [comment]
@ann0see  
I think it should be "src/translation/translation_ko_Kr.qm" changed to "src/translation/translation_ko_KR.qm" in Line 693 in https://github.com/jamulussoftware/jamulus/pull/2685/commits/f4176b2f6ae5c05416825236ef978a259b1def51

**Comment by ann0see** [comment]
Eh. Yes. You should be able to change that too. (Only on my phone at the moment)

**Comment by ann0see** [comment]
Ok. Can you please now check the app if everything looks ok?

**Comment by bagjunggyu** [comment]
It looks Good!

**Comment by bagjunggyu** [comment]
I'm awating other user and musicians feedback now.
If it is OK for them, Let me tell you.
but, for me? it's OK now, good to go.

**Comment by bagjunggyu** [comment]
Just edited by accepting feedback from Korean users and, 
It is their (one of users) actual running image of Jamulus through macOS.
IMG_8648

**Comment by bagjunggyu** [comment]
Now, I edited all. 
This is the end.
Thank you all of you :)

**Comment by ann0see** [comment]
Great! 

To merger: this should be squash merged.

**Comment by bagjunggyu** [comment]
@pljones Should I affect it here as well? 
https://github.com/jamulussoftware/jamulus/pull/2710
https://github.com/jamulussoftware/jamulus/issues/2695

* line 737 source and translation to "Now a directory"
* line 648 source and translation to "Directory server list full"

**Comment by pljones** [comment]
> @pljones Should I affect it here as well?

Thanks, yes.  Can you refresh your master from upstream, rebase and take the necessary steps, please.

**Comment by ignotus666** [comment]
It looks ok to me.

**Comment by bagjunggyu** [comment]
@ignotus666 Thank you :)

**Review by ann0see — CHANGES_REQUESTED** [review]
- [ ] Translator listed in the `src/util.cpp` **optionally add link to PR or code** (No)
- [ ] Punctuation and spacing consistent (todo)
- [ ] Signal words consistent ("ASIO", "Buffer") (todo)
- [x] No untranslated strings (`grep unfinished -5 src/res/translation/translation_$TRANSLTION*.ts`)(please re-check)
- [x] Only a single `.ts` file checked in (+ Installer translations)

**Comment by pljones** [comment]
Plus
* [ ] Please (pull upstream master) and _rebase_, squashing all your changes to a single one on your branch.

**Comment by ann0see** [comment]
I've now rebased and pushed it. Let's hope I pushed the correct version

**Comment by ann0see** [comment]
@gilgongo I already asked:

https://github.com/jamulussoftware/jamulus/pull/2685#issuecomment-1176318060
However, we should still add him to the translators group on GitHub probably?

**Comment by bagjunggyu** [comment]
Thank you very much all of you.
Actually, I'm not a developer like you guys, until now to me, github was just a memo pad :)
I'm trying to rebase as following this manual

"git remote add upstream git@github.com:jamulussoftware/jamulus ;# add the main repo as upstream remote if you didn’t already as mentioned above
git fetch upstream ; # Get the latest changes from the main upstream repo you added before (if needed)
git checkout <theBranchYourWorkIsOn = "patch-1"  in this case right?> ;# switch to (= checkout) the branch you want to rebase
git rebase upstream/master ;# replay (=rebase) your changes onto the latest code (and fix conflicts if needed)
git rebase --continue # to continue the rebasing progress after having fixed the conflicts
git push --force # Push the changes to your repo. Be aware that this will overwrite your remote branch!"

is it right?

**Comment by bagjunggyu** [comment]
@gilgongo Anyway, Thank you very much for your kind inviting me :)

File: src/util.cpp [hunk 3]
```diff
@@ -729,14 +729,15 @@ CAboutDlg::CAboutDlg ( QWidget* parent ) : CBaseDlg ( parent )
                               "<p><b>" +
                               tr ( "Slovak" ) +
                               "</b></p>"
-                              "<p>Jose Riha (<a href=\"https://github.com/jose1711\">jose1711</a>)</p>" +
-                              "<p><b>" + tr ( "Simplified Chinese" ) +
+                              "<p>Jose Riha (<a href=\"https://github.com/jose1711\">jose1711</a>)</p>"
```
Comment (ann0see) [inline src/util.cpp]: This was almost correct. You need to add yourself without a "+" similar to the translators above.

Comment (bagjunggyu) [reply src/util.cpp]: Thanks :) I would try again.

**Comment by bagjunggyu** [comment]
@pljones 
Should I do command "s" for "squash after "git rebase -i master" ?

**Comment by bagjunggyu** [comment]
@ann0see It means already rebased? and nothing left to do?

**Comment by ann0see** [comment]
You still need to squash!

**Comment by bagjunggyu** [comment]
@ann0see 
What about this?
* close this PR
* delete my repository and patch-1
* create new repository and New PR?
I have all things as files.
translation_ko_KR.ts
translation_ko_KR.qm
util.cpp
jamulus.pro

**Comment by bagjunggyu** [comment]
@ann0see 
Never mind, Thank you :)
(Actually, I have failed "git push --force" now several times...)

**Comment by pljones** [comment]
Approving despite my utter lack of Korean!  Thanks!

**Comment by bagjunggyu** [comment]
@pljones Thank you very much :)

**Pull request merged**